include/dav1d/picture.rs

```diff
@@ -207,6 +221,12 @@ unsafe impl AsMutPtr for Rav1dPictureDataComponentInner {
 pub struct Rav1dPictureDataComponent(DisjointMut<Rav1dPictureDataComponentInner>);
 
 impl Rav1dPictureDataComponent {
+    pub fn wrap_buf<BD: BitDepth>(buf: &mut [BD::Pixel], stride: usize) -> Self {
```

Comment:
We're not tying the lifetime of the new `Rav1dPictureDataComponent{Inner}` to the lifetime of the buffer, right? Could we add a lifetime to `Rav1dPictureDataComponent` so that we can ensure that it doesn't outlive the borrowed buffer? The lifetime could be `'static` for owned picture data components. I suspect that'd make the type much more cumbersome to work with though, since we'd need to add lifetime annotations pretty much everywhere this is used.

If that won't work, should we at least add a safety comment here saying that the `Rav1dPictureDataComponent` can't outlive `buf`? I feel like that should make this function `unsafe` as well, since it creates a borrow that's not tracked by the compiler.

Reply:
Yeah, the lifetime is incorrect/unchecked right now.  @rinon brought that up, too.  I think we can add a lifetime that defaults to `'static`.  Can I add that as a PR on top of everything, though, since there's a lot on top of this?

Reply:
Yeah, I'm fine doing that in a separate PR 👍